### ISSUE TITLE
[Question] How to register methods in template class in GDNative?!

### ISSUE THREAD

**Issue by xDGameStudios** [GodotNativeTools/godot-cpp#256]
I can't expose template classes to Godot and this seems fair as it is not supported by the GDScript language. But I wanted to know if there is a way to register methods that are defined within a templated class, for example. How should I proceed?

Here is a code example: `TGraphComponent<G, H>` and `TArc<V>` have methods to be exposed to the GDScript API (but the classes itself are no to be exposed) and `Arc` is a empty class acting as a dummy just to be exposed to the GDScript api.

```cpp
template <typename G, typename H>
class TGraphComponent {

    public:
        int get_id() const;
        /// ...
}

template <typename V>
class TArc : public TGraphComponent<ListDigraph, LemonArc> {

    public:
        Ref<V> get_source() const;
        /// ...
}

class Arc : public Reference, public TArc<Vertex> {
        GODOT_CLASS(Arc, Reference);
}
```

the only way I got it to work is by reimplementing a redirect to the methods in the `Arc` class:

1) WORKING
```cpp
class Arc : public Reference, public TArc<Vertex> {
        GODOT_CLASS(Arc, Reference);

        int get_id() const {
                return TArc::get_id();
        }
        Ref<Vertex> get_source() const {
                return TArc::get_source();
        }
        /// ...

        static void _register_methods();
}
```

then registering the methods as being Arc's in its `_register_methods` function. Even though this works some a little too hacky and there should be way to do this. maybe something in the way register_method works that needs to be changed?!

I more logic way that could be done but DOESN'T WORK, is the one below. All the classes stay the same except for the `Arc` that registers the methods as their own:

2) NOT WORKING
```cpp
class Arc : public Reference, public TArc<Vertex> {
        GODOT_CLASS(Arc, Reference);

        static void _register_methods() {
                register_method("get_id", &Arc::get_id);
                register_method("get_source", &Arc::get_id);
        }
}
```

Is there way to do this?! If so how can I do it? If not then it could be a feature to add. Don't know how much of a change it would require but seems pretty "simple" (I'm no C++ expert so don't judge me 😄). It's legal C++ code and we are not trying to expose the templated classes. And it's nothing that is not supported completely as proven by the workaround on example `1)`.

**Comment by xDGameStudios** [comment]
@BastiaanOlij sorry for tagging you, but since you seam to have a lot of knowledge on C++ and trying to relieve some work from karroffel. hope you could drop some of your knowledge :) thank you in advance

**Comment by raymoo** [comment]
I don't think it even works to give a member function pointer from a non-template base class.

**Comment by raymoo** [comment]
Does trying the workaround in https://github.com/GodotNativeTools/godot-cpp/issues/303#issuecomment-500611318 help?
